Add spancat_singlelabel pipeline for multiclass and non-overlapping span labelling tasks (#11365)
explosion/spaCy: website/docs/api/spancategorizer.md

| `scorer`       | The scoring method. Defaults to [`Scorer.score_spans`](/api/scorer#score_spans) for `Doc.spans[spans_key]` with overlapping spans allowed. ~~Optional[Callable]~~                                                                                                                                       |
| `negative_weight`    | Multiplier for the loss terms. It can be used to downweight the negative samples if there are too many. It is only available for the `spancat_exclusive` component. ~~float~~                                                  |
| `allow_overlap`        | If `True`, the data is assumed to contain overlapping spans. It is only available for the `spancat_exclusive` component. ~~bool~~ |

```python
%%GITHUB_SPACY/spacy/pipeline/spancat.py
```

```python
%%GITHUB_SPACY/spacy/pipeline/spancat_exclusive.py
```

## SpanCategorizer.\_\_init\_\_ {#init tag="method"}

> #### Example
>
> ```python
> # Construction via add_pipe with default model

> > # Use 'spancat_exclusive' for exclusive classes

Reviewer: At first glance this looks like the description is incorrect, maybe something like?

```suggestion
> # Replace 'spancat' with 'spancat_exclusive' for exclusive classes
```

> nlp.add_pipe("spancat", config=config)
> ```

> #### Example (spancat_exclusive)
>
> ```python
> from spacy.pipeline.spancat import DEFAULT_SPANCAT_MODEL
> config = {
>     "threshold": 0.5,
>     "spans_key": "labeled_spans",

> >     "max_positive": None,

Reviewer: `max_positive` isn't used here, is it?

| `threshold`    | Minimum probability to consider a prediction positive. Spans with a positive prediction will be saved on the Doc. Defaults to `0.5`. ~~float~~                                                                                                                                                          |

> | `max_positive` | Maximum number of labels to consider positive per span. Defaults to `None`, indicating no limit. ~~Optional[int]~~                                                                                                                                                                                      |

Reviewer: This would also need a note that it's only for `spancat`.

| `scorer`       | The scoring method. Defaults to [`Scorer.score_spans`](/api/scorer#score_spans) for `Doc.spans[spans_key]` with overlapping spans allowed. ~~Optional[Callable]~~                                                                                                                                       |

> | `negative_weight`    | Multiplier for the loss terms. It can be used to down weigh the negative samples if there are too many. It is only available when using the `spancat_exclusive` component. ~~float~~                                                  |

Reviewer: ```suggestion
| `negative_weight`    | Multiplier for the loss terms. It can be used to downweight the negative samples if there are too many. It is only available for the `spancat_exclusive` component. ~~float~~                                                  |
```

> | `allow_overlap`        | If `True`, the data is assumed to contain overlapping spans. It is only available when using the `spancat_exclusive` component. ~~bool~~ |

Reviewer: ```suggestion
| `allow_overlap`        | If `True`, the data is assumed to contain overlapping spans. It is only available for the `spancat_exclusive` component. ~~bool~~ |
```

```python
%%GITHUB_SPACY/spacy/pipeline/spancat.py
```

```python
%%GITHUB_SPACY/spacy/pipeline/spancat_exclusive.py
```

## SpanCategorizer.\_\_init\_\_ {#init tag="method"}

> #### Example
>
> ```python
> # Construction via add_pipe with default model

> > # Use 'spancat_exclusive' for exclusive clases

Reviewer: ```suggestion
> # Use 'spancat_exclusive' for exclusive classes
```

(And I'm not sure users will understand what "for exclusive classes" means here.)
